### PR TITLE
[ISSUE #9603] Catch ClosedWatchServiceException after exit  normally.

### PR DESCRIPTION
**Pull request by karsonto**
Fix issue #9603 

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java [hunk 1]
```diff
@@ -227,7 +228,7 @@ public void run() {
                             }
                         }
                     });
-                } catch (InterruptedException ignore) {
+                } catch (InterruptedException | ClosedWatchServiceException ignore) {
```
Comment (YunWZ) [inline sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java:231]: 在此处捕获这两个异常时为何要设置中断信号呢？我认为线程应该处理自身的中断信号，而不是仅仅设置其中断标志，且在此处的使用场景中，并没有其他地方会处理该线程的中断信号。

Comment (karsonto) [reply sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java:231]: 你意思是这个异常要单独处理？

Comment (karsonto) [reply sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java:231]: 已修改代码，请再看看

Comment (YunWZ) [reply sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java:231]: 我觉得这样可以。

Comment (KomachiSion) [reply sys/src/main/java/com/alibaba/nacos/sys/file/WatchFileCenter.java:231]: 什么时候会抛出这个异常？

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**